plugins/out_stackdriver/stackdriver_conf.c

```diff
@@ -40,14 +81,38 @@ static inline int key_cmp(const char *str, int len, const char *cmp) {
     return strncasecmp(str, cmp, len);
 }
 
-static int validate_resource(const char *res)
+static int set_resource(const char *res, struct flb_stackdriver *ctx)
 {
-    if (strcasecmp(res, "global") != 0 &&
-        strcasecmp(res, "gce_instance") != 0) {
-        return -1;
+    int i = 0;
+    if (res == NULL) {
+        flb_plg_debug(ctx->ins, "Setting resource as global");
+        ctx->resource = &valid_resources[0];
+        return 0;
+    }
+    while(valid_resources[i].type != NULL) {
+        if (strncmp(valid_resources[i].type, res, strlen(res)) == 0) {
+            flb_plg_debug(ctx->ins,
+                "Setting resource as %s", valid_resources[i].type);
+            ctx->resource = &valid_resources[i];
```

Comment:
Hmm, it's a bit weird that you're actually storing pointers from a static array in the resource list. It happens to work because you only have one instance of `out_stackdriver`, but if you had multiple, with different configurations, this would break. How about storing only the resource schemas (i.e., the list of labels for each resource) in `valid_resources`, and constructing the actual resource structs dynamically?

Reply:
That sounds reasonable. I was just trying to be a minimal on resources as possible. Constructing things dynamically would be fine though, so I will switch to doing that.